### PR TITLE
feat: add wishlist page and it's functionality

### PR DESCRIPTION
**Pull request by devdeadviz**
# Change Logs
- Wishlist page added.
- Add to wishlist and remove from wishlist added.
## Deploy Preview Link:https://deploy-preview-7--meck-store-app.netlify.app/

### CODE REVIEW

**Review by sahiba0915 — APPROVED** [review]
Good use of components, nice that you used usenavigate, properly structured and code looks clean. Good to go.

File: src/components/ProductsCard/ProductsCard.js [hunk 1]
```diff
@@ -35,13 +42,21 @@ const ProductsCard = ({ title, image, price, id, addToCartHandler }) => {
...
             type="button"
-            className="btn btn-primary add-btn my-4 mx-2"
+            className="btn btn-primary add-btn my-1 mx-2"
             onClick={addToCartHandler}
           >
             <i className="fas fa-shopping-cart mr-2"></i>
             Add to Cart
           </button>
         )}
+        <button
```
Comment (ivinayakg) [inline src/components/ProductsCard/ProductsCard.js:52]: try to find a way so that this button can only be clicked once and then it disables or hide

**Review by ivinayakg — APPROVED** [review]
code is good structured and most importantly it is self-explanatory which is great.

File: src/components/ProductsCard/ProductsCard.js (hunk 1 above)
Comment (devdeadviz) [reply src/components/ProductsCard/ProductsCard.js:52]: Yeah, I am thinking about it.

File: src/components/Navbar/Navbar.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
 import { Link } from "react-router-dom";
-import { useCart } from "../../contexts";
+import { useCart, useWishlist } from "../../contexts";
 
 const Navbar = () => {
   const { cartItems } = useCart();
+  const { wishlistItems } = useWishlist();
 
   return (
     <div>
```
Comment (pruthvirajmv) [inline src/components/Navbar/Navbar.js:10]: please check is this div required

File: src/components/ProductsCard/ProductsCard.css [hunk 3]
```diff
@@ -29,8 +29,13 @@
   aspect-ratio: 5/3;
 }
 
-.add-btn {
+.add-btn, .wishlist-btn {
```
Comment (pruthvirajmv) [inline src/components/ProductsCard/ProductsCard.css:32]: good thing

File: src/components/ProductsCard/ProductsCard.js [hunk 4]
```diff
@@ -35,13 +42,21 @@ const ProductsCard = ({ title, image, price, id, addToCartHandler }) => {
         ) : (
           <button
             type="button"
-            className="btn btn-primary add-btn my-4 mx-2"
+            className="btn btn-primary add-btn my-1 mx-2"
             onClick={addToCartHandler}
           >
             <i className="fas fa-shopping-cart mr-2"></i>
             Add to Cart
           </button>
```
Comment (pruthvirajmv) [inline src/components/ProductsCard/ProductsCard.js:50]: suggestion is to save the item's cart status in a variable & use that to display button's content & onClick function

File: src/pages/Cart/Cart.js [hunk 5]
```diff
@@ -43,10 +44,27 @@ const Cart = () => {
...
   return (
     <>
       <Navbar />
       <h1 className="cart-heading text-center">MY CART ({cartItems.length})</h1>
+      {cartItems.length < 1 && (
+        <h2 className="text-center my-4"> Your Cart Is Empty! </h2>
+      )}
       <section className="flex flexJustifyCenter mb-5">
         <section className="flex flexCol flexAlignItemsCenter">
           {cartItems.map(({ title, price, image, _id, qty }) => (
```
Comment (pruthvirajmv) [inline src/pages/Cart/Cart.js:70]: check if you can use ternary operator here

**Review by pruthvirajmv — COMMENTED** [review]
Hey Kuldeep,
Overall the project looks good
Please go through the comments & try to resolve
Also do not forget fix the UI issues like footer & other things as discussed
Keep coding!

**Pull request merged**